src/Upgrade/Infrastructure/VersionControlSystem/SourceCodeProvider/GitLab/GitLabSourceCodeProvider.php

```diff
@@ -131,12 +131,30 @@ protected function mergePullRequest(int $pullRequestId): void
     {
         sleep($this->configurationProvider->getGitLabDelayBetweenPrCreatingAndMerging());
         $this->gitLabClientFactory->getClient()->mergeRequests()->merge(
-            $this->configurationProvider->getGitLabProjectId(),
+            $this->getProjectId(),
             $pullRequestId,
             [
                 'should_remove_source_branch' => true,
                 'merge_when_pipeline_succeeds' => true,
             ],
         );
     }
+
+    /**
+     * @return string
+     */
+    protected function getProjectId(): string
+    {
+        $gitLabProjectId = $this->configurationProvider->getProjectId();
+
+        if ($gitLabProjectId !== '') {
+            return $gitLabProjectId;
+        }
+
+        return sprintf(
+            '%s/%s',
+            $this->configurationProvider->getOrganizationName(),
+            $this->configurationProvider->getRepositoryName(),
+        );
```

Comment:
Does it work like that - without additional request to get the project id?

Reply:
yeap. don't need to request project_id